Cannot compile a channel option within a subcommand

### Issue description

This issue is with TypeScript:

1. Use the code sample below
2. Compile it

Output:
```
source/test.ts:23:13 - error TS2322: Type 'ApplicationCommandOptionTypes.CHANNEL' is not assignable to type 'CommandOptionChoiceResolvableType | CommandOptionNonChoiceResolvableType'.

23             type: Constants.ApplicationCommandOptionTypes.CHANNEL,
               ~~~~

  node_modules/discord.js/typings/index.d.ts:3123:3
    3123   type: CommandOptionNonChoiceResolvableType;
           ~~~~
    The expected type comes from property 'type' which is declared here on type 'ApplicationCommandNonOptionsData | ApplicationCommandChoicesData'


Found 1 error.
```

### Code sample

```typescript
import { Client, Constants, Intents } from "discord.js";

const client = new Client({
  intents: [
    Intents.FLAGS.GUILDS
  ]
});

client.on(Constants.Events.CLIENT_READY, () => {
  const guild = client.guilds.resolve("guild_id");
  if (!guild) throw "No guild.";

  guild.commands.create({
    name: "test",
    description: "test",
    options: [
      {
        type: Constants.ApplicationCommandOptionTypes.SUB_COMMAND,
        name: "test",
        description: "test",
        options: [
          {
            type: Constants.ApplicationCommandOptionTypes.CHANNEL, // Error
            name: "channel",
            description: "A channel.",
            required: false
          }
        ]
      }
    ],
    defaultPermission: true
  });
});

client.login();
```


### discord.js version

13.2.0-dev

### Node.js version

16.6.1. TypeScript: 4.3.5

### Operating system

macOS Big Sur 11.6

### Priority this issue should have

Medium (should be fixed soon)

### Which partials do you have configured?

No Partials

### Which gateway intents are you subscribing to?

GUILDS

### I have tested this issue on a development release

3b14883

--- a/typings/index.d.ts
+++ b/typings/index.d.ts
@@ -3111,12 +3111,12 @@ export interface ApplicationCommandSubGroup extends BaseApplicationCommandOption
 
 export interface ApplicationCommandSubCommandData extends BaseApplicationCommandOptionsData {
   type: 'SUB_COMMAND' | ApplicationCommandOptionTypes.SUB_COMMAND;
-  options?: (ApplicationCommandChoicesData | ApplicationCommandNonOptionsData)[];
+  options?: (ApplicationCommandChoicesData | ApplicationCommandNonOptionsData | ApplicationCommandChannelOptionData)[];
 }
 
 export interface ApplicationCommandSubCommand extends BaseApplicationCommandOptionsData {
   type: 'SUB_COMMAND';
-  options?: (ApplicationCommandChoicesOption | ApplicationCommandNonOptions)[];
+  options?: (ApplicationCommandChoicesOption | ApplicationCommandNonOptions | ApplicationCommandChannelOption)[];
 }
 
 export interface ApplicationCommandNonOptionsData extends BaseApplicationCommandOptionsData {

--- a/typings/tests.ts
+++ b/typings/tests.ts
@@ -1,6 +1,8 @@
 import { APIInteractionGuildMember } from 'discord-api-types';
 import {
   ApplicationCommand,
+  ApplicationCommandChannelOption,
+  ApplicationCommandChannelOptionData,
   ApplicationCommandChoicesData,
   ApplicationCommandData,
   ApplicationCommandManager,
@@ -752,9 +754,10 @@ declare const applicationSubCommandData: ApplicationCommandSubCommandData;
   assertType<'SUB_COMMAND' | ApplicationCommandOptionTypes.SUB_COMMAND>(applicationSubCommandData.type);
 
   // Check that only subcommands can have no subcommand or subcommand group sub-options.
-  assertType<(ApplicationCommandChoicesData | ApplicationCommandNonOptionsData)[] | undefined>(
-    applicationSubCommandData.options,
-  );
+  assertType<
+    | (ApplicationCommandChoicesData | ApplicationCommandNonOptionsData | ApplicationCommandChannelOptionData)[]
+    | undefined
+  >(applicationSubCommandData.options);
 }
 
 declare const guildApplicationCommandManager: GuildApplicationCommandManager;

